### PR TITLE
Jolaneffect Adding and linking effects

### PR DESCRIPTION
**Pull request by Jolan-Goulin**

### CODE REVIEW

File: src/is0/RenderEffect_Edges.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#include "RenderEffect_Edges.h"
+std::string code_gen_edges_parameter(const RenderEffect_Edges& edges)
+{
+    return "const float edges_size = " + std::to_string(*edges.size) +
+           ";\nconst vec3 edges_color = " +
```
Comment (Jolan-Goulin) [inline src/is0/RenderEffect_Edges.cpp:5]: No need to break line

File: src/is0/RenderEffect_Edges.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+std::string code_gen_edges()
+{
+    return R"(
+    if(res.edge == true){
+        finalCol = edges_color ;
+    }
+    )";
+};
+
+bool Edges_imgui(RenderEffect_Edges& edges)
```
Comment (Jolan-Goulin) [inline src/is0/RenderEffect_Edges.cpp:19]: snake_case

File: src/is0/RenderEffect_InigoAO.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
+#include "RenderEffect_InigoAO.h"
+
+std::string code_gen_InigoAO_parameter(const RenderEffect_InigoAO& InigoAO)
```
Comment (Jolan-Goulin) [inline src/is0/RenderEffect_InigoAO.cpp:3]: snake_case

File: src/is0/RenderEffect_InigoAO.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+{
+    return "const float InigoAO_size = " + std::to_string(*InigoAO.size) + ";\n\n";
+};
+
+std::string code_gen_InigoAO_fct()
```
Comment (Jolan-Goulin) [inline src/is0/RenderEffect_InigoAO.cpp:8]: no abbreviations

File: src/is0/RenderEffect_SSAO.cpp [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        vec3(0.909604f, 0.409406f, 0.475478f),
+        vec3(0.0169683f, 0.193884f, 0.221442f),
+        vec3(0.388836f, 0.962859f, 0.175787f),
+        vec3(0.160375f, 0.788018f, 0.0573443f),
+        vec3(0.908658f, 0.969634f, 0.0231941f),
+        vec3(0.749901f, 0.74514f, 0.80697f),
+        vec3(0.943968f, 0.634449f, 0.644398f),
+        vec3(0.787225f, 0.353526f, 0.0561541f)};
+
+    vec3 rotateAxe(vec3 pts, vec3 axe, float angle)
```
Comment (Jolan-Goulin) [inline src/is0/RenderEffect_SSAO.cpp:45]: axe -> axis

Comment (Jolan-Goulin) [inline src/is0/RenderEffect_SSAO.cpp:45]: Use https://suricrasia.online/demoscene/functions/ instead

File: src/is0/RenderEffect_SSAO.cpp [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    {
+        axe     = normalize(axe);
+        float c = cos(angle);
+        float s = sin(angle);
+        mat3  M = mat3(vec3(axe.x * axe.x * (1 - c) + c, axe.x * axe.y * (1 - c) + axe.z * s, axe.x * axe.z * (1 - c) - axe.y * s),
+                    vec3(axe.x * axe.y * (1 - c) - axe.z * s, axe.y * axe.y * (1 - c) + c, axe.y * axe.z * (1 - c) + axe.x * s),
+                    vec3(axe.x * axe.z * (1 - c) + axe.y * s, axe.y * axe.z * (1 - c) - axe.x * s, axe.z * axe.z * (1 - c) + c));
+        return pts * M;
+    }
+    vec3 prodVect(vec3 v, vec3 u)
```
Comment (Jolan-Goulin) [inline src/is0/RenderEffect_SSAO.cpp:55]: Use the native cross function

File: src/is0/RenderEffect_SSAO.cpp [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        }
+        return bl / 32;
+    }
+    )";
+};
+
+std::string code_gen_SSAO()
+{
+    return R"(
+    finalCol = finalCol * ssao(p);
```
Comment (Jolan-Goulin) [inline src/is0/RenderEffect_SSAO.cpp:84]: replace with *=

**Comment by JulesFouchy** [comment]
Merged through other branches

**Pull request closed without merging**